src/variables.ts

```diff
@@ -0,0 +1,37 @@
+import { CustomVariableSupport, DataFrameView, DataQueryRequest, MetricFindValue } from '@grafana/data';
+import { QueryEditor } from 'components/query/QueryEditor';
+import { HierarchyInfo } from './queryResponseTypes';
+
+import { DataSource } from 'DataSource';
+import { QueryType, SitewiseQuery } from 'types';
+import { mergeMap } from 'rxjs/operators';
+import { of } from 'rxjs';
+
+export class SitewiseVariableSupport extends CustomVariableSupport<DataSource, SitewiseQuery> {
+  constructor(private readonly datasource: DataSource) {
+    super();
+  }
+
+  editor = QueryEditor;
+
+  query = (request: DataQueryRequest<SitewiseQuery>) => {
+    const { targets } = request;
+    if (targets && targets.length === 1) {
+      const query = targets[0];
+      if (query.queryType === QueryType.ListAssets || query.queryType === QueryType.ListAssetModels) {
+        return this.datasource.query(request).pipe(
+          mergeMap((rsp) => {
+            const assets = new DataFrameView<HierarchyInfo>(rsp.data[0]);
+            const data: MetricFindValue[] = assets.map((a) => ({
```

Comment:
The default behavior just uses the string values as the variables... BUT for this, I assume we actually want the ids as the template value and the name as what is displayed

Reply:
lgtm

Reply:
We will also need to fetch property IDs. (See thread in conversation)